logic/src/commonMain/kotlin/com/wire/kalium/logic/data/message/mention/MessageMentionMapper.kt

```diff
@@ -57,10 +57,10 @@ class MessageMentionMapperImpl(
     override fun fromProtoToModel(mention: Mention): MessageMention {
         val userId = mention.qualifiedUserId?.let {
             idMapper.fromProtoUserId(it)
-        } ?: UserId(
-            mention.mentionType?.value as String,
-            selfUserId.domain
-        )
+        } ?: run {
+            val id = if (mention.mentionType != null) mention.mentionType?.value as String else ""
```

Comment:
can we maybe ignore the message in case it have invalid proto.
adding empty IDs can lead to unknown behaviors in the future that can be hard to detect.

Reply:
No It's not invalid, it can be received in this way as mention_type and qualified_user_id are optional
```
message Mention {
  required int32 start = 1; // offset from beginning of the message counting in utf16 characters
  required int32 length = 2;
  oneof mention_type {
    // deprecated. Should be set such that old clients always fail when looking
    // up the user. Ideally, this should not be a problem, as a non-federation
    // aware user should never be part of a federated conversation.
    string user_id = 3;
  }
  // only optional to maintain backwards compatibility.
  optional QualifiedUserId qualified_user_id = 4;
}
```